### DISCUSSION TITLE
BEER CSS ANIMATIONS

### DISCUSSION THREAD

**Discussion by im-crooked** [beercss/beercss#364] [General]
Whats the best way to add animations with beercss ?

**Comment by RobinManoli** [comment]
It would be nice to have some basic animations in beer css. For example if you want to display a snackbar, tooltip, or dialog temporarily. Normally this type of message is displayed if you press a copy-to-clipboard button, or after your ajax call is completed.

Not sure if this should be used as a class system: class="fadeout-200ms" or in js .fadeOut() like jquery. Well, since animations are already there for dialogs, maybe there is already a way to use them with js?

Perhaps it's better to just use a animation library?

**Comment by leonardorafael** [comment]
@im-crooked , @RobinManoli 

Beer CSS does the basic transitions only. The best way to do animations using Beer CS is with the `page` element, You need to wrap your element with `page`. Other elements have your own animation like `dialog`, `snackbar`, `menu`, etc. See this codepen https://codepen.io/leo-bnu/pen/zYVMOYG

Example 1:

```
<div class="page left active">
  <...></...>
</div>
```

- the `active` helper will show the content and do the animation
- the `left` helper is where the animation will be start from

Example 2:

```
<div style="display: none">
  <div class="page left active">
    <...></...>
  </div>
</div>
```

- when the container become visible, it will run the animation

Example 3:

```
<button data-ui="#page1">Page 1</button>
<button data-ui="#page2">Page 2</button>
<button data-ui="#page3">Page 3</button>

<div id="page1" class="page right">...</div>
<div id="page2" class="page right">...</div>
<div id="page3" class="page right">...</div>
```

- when you use `data-ui` or `ui()` to trigger a `page` element, all other `page` elements in the same container will be hidden by default

**Reply by lighth7015** [reply]
Beer needs more robust transitions, though, lol.
